File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/java/non-vul/code_before12.java
// From cwe-snippets, snippets_90/non-compliant/Java/cwe-0470/connect_tcp_06.java

private void func() throws Throwable
{
    String data;
    if (PRIVATE_STATIC_FINAL_FIVE != 5)
    {
        data = null;
    }
    else
    {

        /* FIX: Use a hardcoded class name */
        data = "Testing.test";

    }

    Class<?> tempClass = Class.forName(data);
    Object tempClassObject = tempClass.newInstance();

    IO.writeLine(tempClassObject.toString()); /* Use tempClassObject in some way */

}

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/cpp/non-vul/code_before12.cpp
// From cwe-snippets, snippets_90/non-compliant/C++/cwe-0672/list_int_01.cpp

static void func()
{
    list<int>  data;
    data.push_back(100);
    data.push_back(0);
    {
        list<int> ::iterator i;
        cout << "The list contains: ";
        for( i = data.begin(); i != data.end(); i++)
        {
            cout << " " << *i;
        }
        cout << endl;
    }
}


